source/kaleidic/lubeck2.d

```diff
@@ -1228,6 +1228,34 @@ Returns: error code from CBlas
 )
     if (algorithm == "gesvd" || algorithm == "gesdd")
 {
+    
+    size_t info;
+    bool lowApiExecuted;
+
+    auto svdresult = svdImpl!(T, algorithm, kind)(a, info, lowApiExecuted, slim);
+    
+    if(lowApiExecuted)
+    {
+        enum msg = (algorithm == "gesvd" ? "svd: DBDSDC did not converge, updating process failed" : "svd: DBDSQR did not converge");
+        enforce!("svd: " ~ msg)(!info);
+    }
+
+    return svdresult; //transposed
+}
+
+@safe pure @nogc nothrow SvdResult!T svdImpl
+(
+    T,
+    string algorithm = "gesvd",
+    SliceKind kind,
+)(
+    auto ref const Slice!(const(T)*, 2, kind) a,
+    out size_t infoResult,
+    out bool lowApiExecuted,
```

Comment:
why this flag is needed?

Reply:
with the original code, enforce was only triggered in the else block of if (m == 0 || n == 0), meaning that no lapack function ran. With that flag it provides the same logic. I was not going to put that flag. Then that idea seemed meaningful.

Reply:
either way we check info. Do you think the flag is redundant?

Reply:
yes

Reply:
Already removed